ui/src/app/pages/workspace/workspace-library.tsx

```diff
@@ -140,19 +162,34 @@ export const WorkspaceLibrary = withUserProfile()
     }
   }
 
+  updatePublishedWorkspaces(featuredWorkspacesLoadedPromise) {
+    const publishedWorkspaces = this.state.workspaceList.filter(ws =>
+      !fp.contains(ws, this.state.featuredWorkspaces)
+    );
+    this.setState({
+      publishedWorkspaces: publishedWorkspaces
+    })
+  }
+
   render() {
     const {profile: {username}} = this.props.profileState;
-    const {currentTab, errorText, featuredWorkspaces, workspaceList,
-      workspacesLoading} = this.state;
+    const {
+      currentTab,
+      errorText,
+      featuredWorkspaces,
+      publishedWorkspaces,
+      workspaceList,
```

Comment:
The linter does not like that I'm declaring a variable here that's unused. I could:
- decide I don't care and merge anyways
- change the name to 'unused', decide I don't care, and merge anyways
- get rid of this entire declaration and access everything via `this.state`
- figure out a way to keep `workspaceList` out of the state despite it being variable
Thoughts?

Reply:
Maybe I'm missing something here, but this is just an expression which is extracting certain values from this.state using Javascript's destructuring syntax. Can you simply remove the workspaceList entry from this destructuring expression?

Reply:
Wouldn't that assign `this.state.workspaceList` to this scope's `workspacesLoading`?

Reply:
It looks like no, it will not, and instead this is syntactic magic for 'go find the value for the key with that name'